botorch/generation/gen.py

```diff
@@ -109,6 +114,14 @@ def f(x):
         loss = -acquisition_function(X_fix).sum()
         # compute gradient w.r.t. the inputs (does not accumulate in leaves)
         gradf = _arrayify(torch.autograd.grad(loss, X)[0].contiguous().view(-1))
+        if np.isnan(gradf).any():
+            msg = (
+                f"{np.isnan(gradf).sum()} elements of the {x.shape} gradient tensor "
```

Comment:
```suggestion
                f"{np.isnan(gradf).sum()} elements of the {x.size} gradient array "
```

same comment as above